### PR TITLE
Updated crepr.py for Issue #51

### PR DESCRIPTION
**Pull request by Sohammhatre10**
### Added the `report_missing` method

The `report_missing` method is a Typer command that inspects the classes in a given Python file to identify and count those without a `__repr__` method as required in https://github.com/cleder/crepr/issues/51

1. **Input Handling**:
   - The function accepts a list of file paths (`files`) as an argument, which corresponds to Python files that need to be analyzed.

2.**Tracking Classes without `__repr__`**:
   - For each class, it checks if the class has a `__repr__` method by calling `get_repr_source`.
   - If the class doesn't have a `__repr__` method, its name is added to the `no_repr_classes` list, and the class count is incremented.

3. **Reporting**:
   - After analyzing all the classes in a file, the function prints a summary showing how many classes in that file are missing a `__repr__` method.
   - If any classes lack a `__repr__`, their names are printed.
   
   ### Usage for `report-missing`
   ```python
   crepr report-missing <your-file-name>
   ```

___

### **PR Type**
enhancement

___

### **Description**
- Introduced a new Typer command `report_missing` to the `crepr` application.
- The command analyzes specified Python files to count and list classes that do not implement a `__repr__` method.
- Provides a summary output indicating the number of classes missing the `__repr__` method or confirms all classes have it.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>crepr.py</strong><dd><code>Add `report_missing` command to check for missing `__repr__` methods</code></dd></summary>
<hr>

crepr/crepr.py

<li>Added a new <code>report_missing</code> command to the application.<br> <li> The command inspects classes in specified Python files to identify <br>those without a <code>__repr__</code> method.<br> <li> Outputs a summary of classes missing the <code>__repr__</code> method or confirms <br>all classes have it.<br>

</details>

  </td>
  <td><a href="https://github.com/cleder/crepr/pull/52/files#diff-abcd7c4a2773c8234cdc05070d0abd10b0245a1a15f470288240c6d3a108f576">+31/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

## Summary by Sourcery

Add a new Typer command `report_missing` to the crepr.py script, which analyzes specified Python files to report classes missing a `__repr__` method.

New Features:
- Introduce the `report_missing` Typer command to identify and count classes in a Python file that lack a `__repr__` method.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new command function, `report_missing`, to analyze specified files for classes lacking a `__repr__` method and report the findings, including error handling and user feedback.

### CODE REVIEW

**Comment by cleder** [comment]
Have a look at the [static tests](https://github.com/cleder/crepr/actions/runs/10991906521/job/30515218526?pr=52)
```
crepr/crepr.py:358:1: CCR001 Cognitive complexity is too high (19 > 7)
crepr/crepr.py:365:89: E501 line too long (89 > 88 characters)
crepr/crepr.py:393:44: FCS100 too complex f-string
crepr/crepr.py:401:89: E501 line too long (95 > 88 characters)
```

File: crepr/crepr.py [hunk 1]
```diff
@@ -354,5 +354,54 @@ def remove(
...
+
+        try:
+            for _, obj in inspect.getmembers(module, inspect.isclass):
+                if not is_class_in_module(obj, module):
+                    continue
+                _, lineno = get_repr_source(obj)
+                if lineno == -1:
+                    no_repr_classes.append((lineno, obj.__name__))
+                class_count += 1
+        except Exception as e:
```
Comment (cleder) [inline crepr/crepr.py:384]: What exception can occur here? I do not like these broad catch-alls and rather prefer error handling specific to the exception.

**Comment by Sohammhatre10** [comment]
> Have a look at the [static tests](https://github.com/cleder/crepr/actions/runs/10991906521/job/30515218526?pr=52)
> 
> ```
> crepr/crepr.py:358:1: CCR001 Cognitive complexity is too high (19 > 7)
> crepr/crepr.py:365:89: E501 line too long (89 > 88 characters)
> crepr/crepr.py:393:44: FCS100 too complex f-string
> crepr/crepr.py:401:89: E501 line too long (95 > 88 characters)
> ```

@cleder How should I test the same without actually committing the changes? Currently working on the changes right now.

File: crepr/crepr.py (hunk 1 above)
Comment (Sohammhatre10) [reply crepr/crepr.py:384]: @cleder I've tried some modifications for making the code more readable, added explicit specific Error Handling and have tried reducing the static tests too.

File: crepr/crepr.py [hunk 2]
```diff
@@ -354,5 +354,76 @@ def remove(
                 f.write("\n".join(src))
 
 
+@app.command()
+def report_missing(files: Annotated[list[pathlib.Path], file_arg]) -> None:
+    """Count and print classes without a __repr__ method in the source code."""
+    for file_path in files:
```
Comment (cleder) [inline crepr/crepr.py:360]: I added a get_modules function in 7750f43180e678a97bc07eee575ca4a7e4d15dfc to consolidate the error reporting on the Errors

**Pull request closed without merging**